File: woodroid/src/main/java/me/gilo/woodroid/repo/order/OrderNoteRepository.java
package me.gilo.woodroid.repo.order;

import me.gilo.woodroid.data.api.OrderNoteAPI;
import me.gilo.woodroid.models.Order;
import me.gilo.woodroid.models.OrderNote;
import me.gilo.woodroid.repo.WooRepository;
import retrofit2.Call;

import java.util.List;

public class OrderNoteRepository extends WooRepository {

    private final OrderNoteAPI apiService;

    public OrderNoteRepository(String baseUrl, String consumerKey, String consumerSecret) {
       super(baseUrl, consumerKey, consumerSecret);
       apiService = retrofit.create(OrderNoteAPI.class);
    }

    public Call<OrderNote> create(Order order, OrderNote note) {
        return apiService.create(order.getId(), note);
    }

    public Call<OrderNote> note( Order order, int id) {
        return apiService.view(order.getId(), id);
    }

    public Call<List<OrderNote>> notes(Order order) {
        return apiService.list(order.getId());
    }

    public Call<OrderNote> delete(Order order, int id) {
        return apiService.delete(order.getId(), id);
    }

    public Call<OrderNote> delete(Order order, int id, boolean force) {
        return apiService.delete(order.getId(), id, force);
    }


}

File: woodroid/src/main/java/me/gilo/woodroid/repo/order/OrderNoteRepository.kt
package me.gilo.woodroid.repo.order

import me.gilo.woodroid.data.api.OrderNoteAPI
import me.gilo.woodroid.models.Order
import me.gilo.woodroid.models.OrderNote
import me.gilo.woodroid.models.filters.OrderNoteFilter
import me.gilo.woodroid.repo.WooRepository
import retrofit2.Call

class OrderNoteRepository(baseUrl: String, consumerKey: String, consumerSecret: String) :
    WooRepository(baseUrl, consumerKey, consumerSecret) {

    private val apiService: OrderNoteAPI

    init {
        apiService = retrofit.create(OrderNoteAPI::class.java)
    }

    fun create(order: Order, note: OrderNote): Call<OrderNote> {
        return apiService.create(order.getId()!!, note)
    }

    fun note(order: Order, id: Int): Call<OrderNote> {
        return apiService.view(order.getId()!!, id)
    }

    fun notes(order: Order): Call<List<OrderNote>> {
        return apiService.list(order.getId()!!)
    }

    fun notes(order: Order, orderNoteFilter: OrderNoteFilter): Call<List<OrderNote>> {
        return apiService.filter(order.getId()!!, orderNoteFilter.filters)
    }

    fun delete(order: Order, id: Int): Call<OrderNote> {
        return apiService.delete(order.getId()!!, id)
    }

    fun delete(order: Order, id: Int, force: Boolean): Call<OrderNote> {
        return apiService.delete(order.getId()!!, id, force)
    }


}
